File: src/net/merayen/elastic/backend/nodes/list/test_100/LogicNode.java
package net.merayen.elastic.backend.nodes.list.test_100;

import net.merayen.elastic.backend.nodes.BaseLogicNode;
import net.merayen.elastic.backend.nodes.Format;
import net.merayen.elastic.system.intercom.NodeParameterMessage;

public class LogicNode extends BaseLogicNode {

	@Override
	protected void onCreate() {
		BaseLogicNode.PortDefinition input = new BaseLogicNode.PortDefinition();
		input.name = "input";
		input.format = new Format[]{Format.AUDIO};
		createPort(input);
	}

	@Override
	protected void onParameterChange(NodeParameterMessage message) {
		set(message);
	}

	@Override
	protected void onConnect(String port) {}

	@Override
	protected void onDisconnect(String port) {}
}

File: src/net/merayen/elastic/backend/logicnodes/list/sample_1/LogicNode.kt
package net.merayen.elastic.backend.logicnodes.list.sample_1

import net.merayen.elastic.backend.logicnodes.Format
import net.merayen.elastic.backend.nodes.BaseLogicNode

class LogicNode : BaseLogicNode() {
	override fun onCreate() {
		createPort(object: PortDefinition() {
			init {
				name = "control"
			}
		})

		createPort(object : PortDefinition() {
			init {
				name = "out"
				format = Format.AUDIO
				output = true
			}
		})
	}

	override fun onInit() {}
	override fun onParameterChange(key: String?, value: Any?) = set(key, value)
	override fun onData(data: MutableMap<String, Any>?) {}
	override fun onConnect(port: String?) {}
	override fun onDisconnect(port: String?) {}
	override fun onRemove() {}
	override fun onPrepareFrame(data: MutableMap<String, Any>?) {}
	override fun onFinishFrame(data: MutableMap<String, Any>?) {}
}